### ISSUE TITLE
Show applicants a library of all documents they've uploaded

### ISSUE THREAD

**Issue by neel-a-mehta** [seattle-uat/civiform#1443]
Although Civiform enables reuse of answers, applicants may still need to re-upload the same document to multiple questions (e.g. use your bank statement for proof of income and proof of residency). 

Finding documents can be a pain, according to the CBOs we've talked to, so we should save the documents that applicants have uploaded and make it easy for them to find those documents again.

We should create a tab named Library (or similar) for applicants and show all the files they've ever uploaded to a File Upload question. Let them download those files or delete them.

We could also have all File Upload questions offer a section to "upload a file from your library", which opens a dropdown with the files you've uploaded to CiviForm before. You can just tap one to upload them.

**Comment by baipai** [comment]
Assigned for ux design part.

**Comment by kristinyim** [comment]
Dependent on #1496

**Comment by elisekalstad** [comment]
@msprenke -- Just flagging this one for your review and comments.

**Comment by msprenke** [comment]
Work started at Exygy on this https://docs.google.com/document/d/1-NoC72SWNyuNOKV61V6QUHaSQgNuKn4hn5O1eWLA4AM/edit?tab=t.0 currently on hold due to other priorities
